File: fix_imports.py
import os
import re

def walk_dir(directory):
    for root, _, files in os.walk(directory):
        if 'node_modules' in root or '.git' in root: continue
        for file in files:
            if file.endswith(('.ts', '.tsx')):
                yield os.path.join(root, file)

import_pattern = re.compile(r'import\s+\{([^}]+)\}\s+from\s+[\'"]([^\'"]*client)[\'"]')

# Mock patterns
# vi.mock('./client', () => ({ isSupabaseConfigured: () => true, ... }))
def fix_file(filepath):
    with open(filepath, 'r') as f:
        content = f.read()

    orig = content
    
    # 1. Fix Imports
    def handle_import(m):
        imports_str = m.group(1)
        mod_path = m.group(2)
        imports = [x.strip() for x in imports_str.split(',')]
        if 'isSupabaseConfigured' not in imports:
            return m.group(0)
            
        others = [x for x in imports if x != 'isSupabaseConfigured' and x]
        config_path = mod_path.replace('client', 'config')
        
        lines = []
        if others:
            lines.append(f"import {{ {', '.join(others)} }} from '{mod_path}'")
        lines.append(f"import {{ isSupabaseConfigured }} from '{config_path}'")
        return '\n'.join(lines)
        
    content = import_pattern.sub(handle_import, content)

    # 2. Fix Mocks
    # Some files mock `isSupabaseConfigured` like:
    # isSupabaseConfigured: () => true
    # isSupabaseConfigured: () => false
    # isSupabaseConfigured: false
    # isSupabaseConfigured: mockIsConfigured
    
    # find where vi.mock passes a string ending in 'client'
    mock_pattern = re.compile(r'(vi\.mock\([\'"][^\'"]*client[\'"],[^)]*?\n\}\)\))', re.DOTALL)
    
    # Actually, a safer way to add the config mock is just to see if the file tests things requiring it, 
    # but the easiest is simply replacing `isSupabaseConfigured: <value>` inside the client mock,
    # and extracting it into a separate config mock right before or after the client mock.
    
    # We can just search for the specific lines:
    mock_val_pattern = re.compile(r'^\s*isSupabaseConfigured:\s*(.+?),?$', re.MULTILINE)
    
    if '.test' in filepath:
        # Check if it has a mock for client
        matches = list(re.finditer(r'vi\.mock\([\'"]([^\'"]*client)[\'"]', content))
        if matches and mock_val_pattern.search(content):
            for m in matches:
                mod = m.group(1)
                config_mod = mod.replace('client', 'config')
                
                # Find the `isSupabaseConfigured` value in the file
                val_match = mock_val_pattern.search(content)
                if val_match:
                    val = val_match.group(1)
                    # remove it from the original content
                    content = content.replace(val_match.group(0), '')
                    
                    # add the new mock
                    new_mock = f"\nvi.mock('{config_mod}', () => ({{\n  isSupabaseConfigured: {val},\n  credentials: null\n}}))\n"
                    # insert right before the client mock
                    idx = content.find(m.group(0))
                    content = content[:idx] + new_mock + content[idx:]

    if content != orig:
        with open(filepath, 'w') as f:
            f.write(content)
        print("Fixed", filepath)

for f in walk_dir('src'):
    fix_file(f)

File: fix_imports.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Walk through directory
function walkSync(dir, filelist = []) {
  fs.readdirSync(dir).forEach(file => {
    const filePath = path.join(dir, file);
    if (fs.statSync(filePath).isDirectory()) {
      if (!filePath.includes('node_modules') && !filePath.includes('.git')) {
        filelist = walkSync(filePath, filelist);
      }
    } else {
      if (filePath.endsWith('.ts') || filePath.endsWith('.tsx') || filePath.endsWith('.js')) {
        filelist.push(filePath);
      }
    }
  });
  return filelist;
}

const files = walkSync(path.join(__dirname, 'src'));

let updatedFilesCount = 0;

files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  let originalContent = content;

  // 1. Update application code imports
  // Look for: import { ..., isSupabaseConfigured, ... } from './client' (or similar path)
  // This is a bit complex with regex. A simpler approach is to find the exact import statement and modify it.
  
  if (content.includes('isSupabaseConfigured')) {
    // If it's importing from a client file:
    const importRegex = /import\s+{([^}]*isSupabaseConfigured[^}]*)}\s+from\s+['"]([^'"]*client(\.js|\.ts)?)['"]/g;
    
    content = content.replace(importRegex, (match, importsStr, modulePath) => {
      // importsStr is like " supabase, isSupabaseConfigured "
      const imports = importsStr.split(',').map(s => s.trim()).filter(Boolean);
      const remainingImports = imports.filter(i => i !== 'isSupabaseConfigured');
      
      const configPath = modulePath.replace(/client(\.js|\.ts)?$/, 'config');
      
      let newResult = '';
      if (remainingImports.length > 0) {
         newResult += `import { ${remainingImports.join(', ')} } from '${modulePath}'\n`;
      }
      newResult += `import { isSupabaseConfigured } from '${configPath}'`;
      return newResult;
    });
  }

  // 2. Update test mocks
  if (file.includes('.test.') && content.includes('vi.mock(')) {
     // Look for vi.mock('./client', ... { isSupabaseConfigured: false })
     // We need to keep the client mock for supabase.from(), and add a config mock for isSupabaseConfigured.
     
     // But wait! Many tests just mock isSupabaseConfigured and don't care about anything else from client.
     // Some tests mock Supabase client methods like `updateUser` AND `isSupabaseConfigured`.
     const mockRegex = /vi\.mock\(['"]([^'"]*client)['"],/g;
     
     // Instead of regex manipulating mocks which is very brittle, we can append a mock for config.
     // Let's do a simple replace: if we see vi.mock('.../client', ...
     // we can just make sure we also mock config if isSupabaseConfigured is used in this file.
     // Let's just do it manually for test files using sed or individual script logic since there aren't *too* many tests failing.
     // Actually, let's just replace `isSupabaseConfigured` in the mock object specifically.
  }

  if (content !== originalContent) {
    fs.writeFileSync(file, content);
    updatedFilesCount++;
  }
});

console.log(`Updated ${updatedFilesCount} files with new imports.`);
